Clean up handling of background-position shorthand in StyleProperties::asTextInternal()

nt1m opened the pull request:
#### fd7fe182b10d940000b88d4ec01f5a33e62df99e
<pre>
Clean up handling of background-position shorthand in StyleProperties::asTextInternal()
<a href="https://bugs.webkit.org/show_bug.cgi?id=247879">https://bugs.webkit.org/show_bug.cgi?id=247879</a>
rdar://102306296

Reviewed by Antti Koivisto.

The old code was collapsing background-position-x/y into background-position.
It is not needed anymore, as returning true in canUseShorthandForLonghand for background-position has the same effect.

Test: fast/css/background-position-serialize.html

* LayoutTests/fast/css/remove-shorthand-expected.txt:
* Source/WebCore/css/StyleProperties.cpp:
(WebCore::canUseShorthandForLonghand):
(WebCore::StyleProperties::asTextInternal const):

Canonical link: <a href="https://commits.webkit.org/256643@main">https://commits.webkit.org/256643@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/abb3b7f102f259d636dae178d031dfc212beaca2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/96400 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/5654 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/29465 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/105943 "Built successfully") | [💥 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/166291 "An unexpected error occured. Recent messages:Pull request contains relevant changes; Failed to print configuration") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/100383 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/5825 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/34400 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/88779 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/102669 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/102071 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/4335 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/83002 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/31318 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/86177 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/88068 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/74209 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/40130 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/19560 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/37804 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/20948 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4610 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/3173 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/43521 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/720 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/40217 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

anttijk reviewed (approved):
That is a lot of red, nice!

nt1m committed:
Clean up handling of background-position shorthand in StyleProperties::asTextInternal()
https://bugs.webkit.org/show_bug.cgi?id=247879
rdar://102306296

Reviewed by Antti Koivisto.

The old code was collapsing background-position-x/y into background-position.
It is not needed anymore, as returning true in canUseShorthandForLonghand for background-position has the same effect.

Test: fast/css/background-position-serialize.html

* LayoutTests/fast/css/remove-shorthand-expected.txt:
* Source/WebCore/css/StyleProperties.cpp:
(WebCore::canUseShorthandForLonghand):
(WebCore::StyleProperties::asTextInternal const):

Canonical link: https://commits.webkit.org/256643@main
